--- 3dcnxplug-win.cpp	2008-10-20 21:33:31.562500000 +0200
+++ Kopie von 3dcnxplug-win.cpp	2008-10-22 11:59:11.328125000 +0200
@@ -180,7 +180,10 @@
 
 			// specify our ID (useful if blender can host multiple plugins 
 			// using our device)
-			cnx_simpledev->LoadPreferences(L"3DxBlender");
+			BSTR preferencesName;
+			preferencesName = SysAllocString(L"Blender");
+			cnx_simpledev->LoadPreferences(preferencesName);
+			SysFreeString(preferencesName);
 		}
 		cnx_dev.Release();
 	}
@@ -284,7 +287,7 @@
 }
 
 HRESULT NDOFServer::
-OnDevKeyDown(int keycode)
+OnDevKeyDown(long keycode)
 {
 	if (m_DevKeyb)
 	{
--- 3dcnxplug-win.h	2007-08-01 14:59:00.000000000 +0200
+++ Kopie von 3dcnxplug-win.h	2008-10-22 11:53:10.703125000 +0200
@@ -56,7 +56,7 @@
     HRESULT OnSensorInput();
 
 	/** Callback invoked by the COM server when device keys are pressed. */
-	HRESULT OnDevKeyDown(int keycode);
+	HRESULT OnDevKeyDown(long keycode);
 
 protected:
     int initCOM();
